plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -64,42 +64,49 @@ var defaultConfig = config.Configuration{
 	LatencyUnit:                 "ms",
 }
 
-func tracerProvider(t *testing.T) (trace.TracerProvider, *tracetest.InMemoryExporter) {
+func tracerProvider(t *testing.T) (trace.TracerProvider, *tracetest.InMemoryExporter, func(ctx context.Context) error) {
 	exporter := tracetest.NewInMemoryExporter()
 	tp := sdktrace.NewTracerProvider(
 		sdktrace.WithSampler(sdktrace.AlwaysSample()),
 		sdktrace.WithSyncer(exporter),
 	)
-	return tp, exporter
+	closer := func(ctx context.Context) error {
+		return tp.Shutdown(ctx)
+	}
+	return tp, exporter, closer
 }
 
 func TestNewMetricsReaderValidAddress(t *testing.T) {
 	logger := zap.NewNop()
-	tracer, _ := tracerProvider(t)
+	tracer, _, closer := tracerProvider(t)
 	reader, err := NewMetricsReader(config.Configuration{
 		ServerURL:      "http://localhost:1234",
 		ConnectTimeout: defaultTimeout,
 	}, logger, tracer)
 	require.NoError(t, err)
 	assert.NotNil(t, reader)
+	err = closer(context.Background())
```

Comment:
closer should be invoked via defer right after the object is created. Otherwise require in L86 may panic and you would never reach this line.

Reply:
I see, I will add defer to it.